drivers/t5838/t5838_nrfx_pdm.c

```diff
@@ -496,53 +1012,87 @@ static void init_clock_manager(const struct device *dev)
 	__ASSERT_NO_MSG(drv_data->clk_mgr != NULL);
 }
 
-static const struct _dmic_ops dmic_ops = {
+static const struct _dmic_ops t5838_dmic_ops = {
 	.configure = t5838_configure,
 	.trigger = t5838_trigger,
 	.read = t5838_pdm_read,
 };
 
-#define PDM(idx)	 DT_NODELABEL(pdm##idx)
-#define PDM_CLK_SRC(idx) DT_STRING_TOKEN(PDM(idx), clock_source)
-
-#define PDM_NRFX_DEVICE(idx)                                                                       \
-	static void *rx_msgs##idx[DT_PROP(PDM(idx), queue_size)];                                  \
+#define T5838_NODE_ID(idx)     DT_NODELABEL(t5838)
+#define T5838_PDM(idx)	       DT_NODELABEL(pdm##idx)
+#define T5838_PDM_CLK_SRC(idx) DT_STRING_TOKEN(T5838_PDM(idx), clock_source)
+
+#define T5838_HAS_THSEL_GPIO(idx) DT_NODE_HAS_PROP(DT_DRV_INST(idx), thsel_gpios)
+#define T5838_THSEL_GPIO_SPEC(idx)                                                                 \
+	IF_ENABLED(T5838_HAS_THSEL_GPIO(idx),                                                      \
+		   (static const struct gpio_dt_spec thsel_##idx =                                 \
+			    GPIO_DT_SPEC_GET(DT_DRV_INST(idx), thsel_gpios);))
+
+#define T5838_HAS_MICEN_GPIO(idx) DT_NODE_HAS_PROP(DT_DRV_INST(idx), micen_gpios)
+#define T5838_MICEN_GPIO_SPEC(idx)                                                                 \
+	IF_ENABLED(T5838_HAS_MICEN_GPIO(idx),                                                      \
+		   (static const struct gpio_dt_spec micen_##idx =                                 \
+			    GPIO_DT_SPEC_GET(DT_DRV_INST(idx), micen_gpios);))
+
+#define T5838_HAS_WAKE_GPIO(idx) DT_NODE_HAS_PROP(DT_DRV_INST(idx), wake_gpios)
+#define T5838_WAKE_GPIO_SPEC(idx)                                                                  \
+	IF_ENABLED(T5838_HAS_WAKE_GPIO(idx),                                                       \
+		   (static const struct gpio_dt_spec wake_##idx =                                  \
+			    GPIO_DT_SPEC_GET(DT_DRV_INST(idx), wake_gpios);))
+
+#define T5838_HAS_PDMCLK_GPIO(idx) DT_NODE_HAS_PROP(DT_DRV_INST(idx), pdmclk_gpios)
+#define T5838_PDMCLK_GPIO_SPEC(idx)                                                                \
+	IF_ENABLED(T5838_HAS_PDMCLK_GPIO(idx),                                                     \
+		   (static const struct gpio_dt_spec pdmclk_##idx =                                \
+			    GPIO_DT_SPEC_GET(DT_DRV_INST(idx), pdmclk_gpios);))
```

Comment:
Why are this `gpio_dt_spec` pins declared here and then passed as pointers to the configuration struct?

Remove this and assign them directly.

Reply:
I did so since same implementation was done in our en25 driver, which I was looking at as an example of implementation. Will change